index.js

```diff
@@ -1,207 +1,184 @@
 'use strict'
 
+const promisify = require('util').promisify
 const fs = require('fs')
 const path = require('path')
-const steed = require('steed')
-
-module.exports = function (fastify, opts, next) {
-  const defaultPluginOptions = opts.options
-
-  const packagePattern = /^package\.json$/im
-  const indexPattern = opts.includeTypeScript
-    ? /^index\.(ts|js)$/im
-    : /^index\.js$/im
-  const scriptPattern = opts.includeTypeScript
-    ? /((^.?|\.[^d]|[^.]d|[^.][^d])\.ts|\.js)$/im // For .ts files, ignore .d.ts
-    : /\.js$/im
-
-  function enrichError (err) {
-    // Hack SyntaxError message so that we provide
-    // the line number to the user, otherwise they
-    // will be left in the cold.
-    if (err instanceof SyntaxError) {
-      err.message += ' at ' + err.stack.split('\n')[0]
+const readDirectory = promisify(fs.readdir)
```

Comment:
We could use `require('fs').promises` instead of promisify just to be stricter (it is supported from node 10+)